src/gp_instance.hpp

```diff
@@ -52,8 +52,14 @@ class GPInstance {
   SizeDoubleVectorMap GatherBranchLengths();
   void TakeFirstBranchLength();
   void EstimateSBNParameters();
+  void SetOptimizationMethod(const OptimizationMethod method);
+  void UseGradientOptimization(const bool use_gradients);
   void EstimateBranchLengths(double tol, size_t max_iter, bool quiet = false,
-                             bool track_intermediate_iterations = false);
+                             bool track_intermediate_iterations = false,
+                             std::optional<OptimizationMethod> method = std::nullopt);
+  void EstimateTPBranchLengths(double tol, size_t max_iter, bool quiet = false,
```

Comment:
It looks like these are only used in tests?

Reply:
Yes, `EstimateTPBranchLength` is just used for testing.  `EstimateBranchLength` is in the pybito API, so it looks like someone is using it elsewhere.